packages/xsnap/src/avaXS.js

```diff
@@ -123,60 +165,42 @@ async function main(
   argv,
   { bundleSource, spawn, osType, readFile, resolve, dirname },
 ) {
-  async function testSource(input) {
-    const bundle = await bundleSource(input, 'getExport', {
-      externals: [
-        'ava',
-        '@agoric/bundle-source',
-        '@agoric/install-ses',
-        '@agoric/install-metering-and-ses',
-      ],
-    });
-    return bundle.source;
-  }
-
-  const sesShim = await asset(`../dist/bootstrap.umd.js`, readFile);
-  const tinyAva = await asset(`./avaAssertXS.js`, readFile);
-  const avaHandler = await asset(`./avaHandler.js`, readFile);
-
-  const debug = extractFlag(argv, '--debug');
-
-  // we only use import() in type annotations
-  const hideImport = (/** @type { string } */ src) =>
-    src.replace(/import\(/g, '');
+  const args = argv.slice(2);
+  const debug = args[0] === '--debug';
+  const files = debug ? args.slice(1) : args;
 
-  async function withXSnap(opts, thunk) {
-    const worker = xsnap({
-      meteringLimit: 0,
+  const spawnXSnap = opts =>
+    xsnap({
+      ...opts,
+      debug,
       spawn,
       os: osType(),
+      meteringLimit: 0,
       stdout: 'inherit',
       stderr: 'inherit',
-      debug,
-      ...opts,
     });
```

Comment:
I notice you switched the `...opts` from end to beginning. These happen in order so later things get to override earlier things. Is this consistent with your intended use of `opts`?

Reply:
yes. main "owns" the options for spawn, debug, etc.